MapboxCoreNavigation/SimulatedLocationManager.swift

```diff
@@ -65,19 +65,40 @@ public class SimulatedLocationManager: NavigationLocationManager {
      */
     @objc public init(route: Route) {
         super.init()
+        initializeSimulatedLocationManager(for: route)
+    }
+
+    /**
+     Initalizes a new `SimulatedLocationManager` with the given route and distance traveled.
+     
+     When location simulation is needed in special cases such as dead reckoning, the distance
+     traveled on the given route determines where the `SimulatedLocationManager` needs to begin the simulation.
+     
+     - parameter route: The initial route.
+     - parameter distanceTraveled: The distance traveled on the current route.
+     - returns: A `SimulatedLocationManager`
+     */
+    @objc public init(route: Route, distanceTraveled: CLLocationDistance) {
+        super.init()
+        self.currentDistance = distanceTraveled
+        initializeSimulatedLocationManager(for: route)
+    }
+
+    private func initializeSimulatedLocationManager(for route: Route) {
         self.route = route
-        reset()
         NotificationCenter.default.addObserver(self, selector: #selector(didReroute(_:)), name: .routeControllerDidReroute, object: nil)
         NotificationCenter.default.addObserver(self, selector: #selector(progressDidChange(_:)), name: .routeControllerProgressDidChange, object: nil)
     }
     
-    private func reset() {
+    private func reset(_ distance: CLLocationDistance) {
         if let coordinates = route?.coordinates {
             routeLine = coordinates
             locations = coordinates.simulatedLocationsWithTurnPenalties()
             
-            currentDistance = 0
+            currentDistance = distance
             currentSpeed = 30
+            
+            currentDistance += currentSpeed * speedMultiplier
```

Comment:
Why are we updating `currentDistance` twice now within this function?

Reply:
This is for switching `LocationManager` between `NavigationLocationManager` and `SimulatedLocationManager`. You need to update the `currentDistance` in this function for the location updates function - `tick`.

Reply:
@bsudekum Yes we need that, to accompany the enhancements made here: https://github.com/mapbox/mapbox-navigation-ios/pull/1218/files/1d6f00b6d7e604dff75a2e57f9e0fab21d2b9ff3#diff-9bf14ebab2a9283285d3a8e91bd09e86R81

Reply:
Ok, let's only mutate distance once then.

Reply:
This is the intent of the code. 
`currentDistance = distance + (currentSpeed * speedMultiplier)`
Will be updated to avoid future confusion.